.github/workflows/code-generation.yaml

```diff
@@ -0,0 +1,71 @@
+# This file is part of Astarte.
+#
+# Copyright 2023 SECO Mind Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: Code generation
+
+on:
+  push:
+  pull_request:
+    types: [closed]
```

Comment:
Is the `closed` event on pull_request also triggered when the PR is closed manually by clicking the `close` button or only when it is merged?

Reply:
Closed is always triggered, also when closed manually. It appears there is no way to trigger the workflow only when the PR is merged.
However, we can further limit the triggering of the commit job/step by using the conditional:
```
if: github.event_name == 'pull_request' && github.event.pull_request.merged == true
```

Reply:
To be added in the still missing `commit` job.